Updates changelog for v1.4.2 (#2037)
l3montree-dev/devguard: CHANGELOG.md

All notable changes to this project will be documented in this file.

## [v1.4.2] - 2026-05-20

### Fixed

- SBOM graph normalization panicked when a component had multiple info-source parents — multi-parent cases are now handled
- Cascade delete for `github_app_installations` and `artifact_license_risks` — installations and license risks are removed when their parent records are deleted (migration uses the correct `license_risk_id` column name)
- **Web:** PDF download path; autosetup loading state on failure

### Added

- `DisablePublicRequest` middleware to enforce the public-request toggle at the route level, with router tests verifying it is applied to the intended endpoints

### Contributors (Web)

[@seb-kw](https://github.com/seb-kw); external: [@resolvicomai](https://github.com/resolvicomai) (Mauro Marques Filho) — autosetup fix

## [v1.4.1] - 2026-05-19

### Fixed

- Nil pointer dereference in `asset_version_service`

> - **Web:** last-active-org redirect: SSR hydration mismatch, localStorage placeholder-org guards, session update on org registration; streaming-chunk buffering for newline-delimited JSON parsing; SBOM/SARIF order in the own-scanner upload flow

Reviewer: ```suggestion
#### API
- Nil pointer dereference in `asset_version_service`

#### Web
- last-active-org redirect: SSR hydration mismatch
- localStorage placeholder-org guards, session update on org registration
- streaming-chunk buffering for newline-delimited JSON parsing
- SBOM/SARIF order in the own-scanner upload flow

```

### Changed

- **Web:** CVE marked optional on vulnerability views (matches backend foreign-key removal); lightmode severity colors adjusted for contrast

### Contributors (Web)

[@timbastin](https://github.com/timbastin), [@refoo0](https://github.com/refoo0), [@juliankepka](https://github.com/juliankepka)

## [v1.4.0] - 2026-05-19

### Added

> - VulnDB v2 — Rewrite of the vulnerability database pipeline. The published vulndb image is now a single streaming bundle of gob-encoded, zst-compressed files (CVEs, affected components, CVE relationships, EPSS, CISA KEV, exploits, malicious packages) instead of multiple data sources fetched at runtime

Reviewer: ```suggestion
- VulnDB v2 — Complete rewrite of the vulnerability database pipeline. The published VulnDB image is now distributed as a single streaming bundle of gob-encoded, Zstandard-compressed datasets (CVEs, affected components, CVE relationships, EPSS, CISA KEV, exploits, and malicious packages), replacing the previous model of fetching multiple data sources at runtime. In addition, every table is checksummed during the GitHub Actions build process, and the resulting artifact embeds metadata containing these checksums. After both quick-diff and full streaming imports, integrity is verified by ensuring the imported state matches the original build output using Merkle-tree-based validation.
```

- Quick-diff incremental updates — VulnDB clients apply only the rows that changed since the last sync via a stage-table EXCEPT-based diff, with a streaming fallback if quick-diff fails and a monitoring alert when it does
- Streaming imports — Streaming transformers pipe gob files into PostgreSQL using buffered channels and bulk inserts; staging tables are flushed once per stream; index rebuild is triggered if the local vulndb is older than 7 days
- Embedded vulndb cosign public key — The cosign pubkey used to verify the vulndb image is embedded in the DevGuard binary; content-hash columns added to malicious packages and exploits for integrity verification
- Crowdsourced VEX — Recommendation algorithm with project-based recommendations, vote keying, VEX rules included in recommendation output, and matching DTOs
- Deep search — Search endpoint that returns projects together with their subprojects and assets in a single query
- Admin instance settings — Endpoint and middleware to read and update instance-level settings; `DISABLE_ORG_CREATION` config option for single-organization deployments

> - OCI proxy hardening — SSRF protection for the public OCI dependency proxy; path-parameter validation; GitLab registry support; public kill switch via the `DisablePublicRequest` middleware

Reviewer: ```suggestion
- OCI proxy hardening — SSRF protection for the public OCI dependency proxy; path-parameter validation; GitLab registry support;
```

- User-agent propagation — User agent threaded through controllers, services, and integrations (events, license decisions, Jira); `user_agent` column on the events table; MCP-server `CreateEvent` calls are tagged accordingly

> - Fixable CVSS counts in risk statistics; risk calculation uses the highest risk per CVE/PURL pair

Reviewer: ```suggestion
- Fixable CVSS counts in risk statistics; risk calculation uses the highest risk per CVE/PURL pair. This builds upon the QuickFix Algoritm (https://docs.devguard.org/explanations/supply-chain-security/transitive-vulnerability-path-analysis/)
```

- Daemon pipeline timeout raised to 2 hours to surface stuck imports instead of blocking the queue
- Integration tests for scoped SBOM scans with artifact-specific vulnerabilities
- OpenTelemetry spans on vulndb `ImportRC` and `checkIfTokenIsValid`, including retry attributes
- Dedicated health-check database connection; db-stats logging on failing health checks

> - `DEBUG_LOCAL_ZIP` support for local vulndb development

Reviewer: ```suggestion
```
